File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const hash = new Map()
        const bucket = Array.from({length: nums.length + 1}, () => [])

        for(let i = 0; i < nums.length; i++){
            hash.set(nums[i], (hash.get(nums[i]) ?? 0) + 1)
        }

        for(const [key, value] of hash.entries()){
            bucket[value].push(key)
        }
        const result = []
        for(let r = bucket.length - 1; r >= 0; r--){
            if(bucket[r].length === 0) continue
            for(let n = bucket[r].length - 1; n >= 0; n--){
                result?.push(bucket[r][n])
                if(result.length === k) return result
            }
        }
        return result
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-5.cpp
class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        vector<int> result = {};
        unordered_map<int, int> m = {};
        int max = -10001;

        for(int i = 0; i < k; i++){
            if(nums[i] > max) max = nums[i];
            m[i]=nums[i];
        }

        result.push_back(max);
        max = -10001;
        int l = 0;
        for(int r = k; r < nums.size(); r++){
            m[r] = nums[r];
            m[l] = -10001;
            l++;
            for(const auto [key, value] : m){
                if(value > max) max = value;
            }
            result.push_back(max);
            max = -10001;
        }
        return result;
    }
};
